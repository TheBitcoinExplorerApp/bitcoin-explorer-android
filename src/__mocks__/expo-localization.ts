const getLocales = () => [
  { country: "pt", languageTag: "pt-BR", languageCode: "pt", isRTL: false },
];

export default getLocales;
