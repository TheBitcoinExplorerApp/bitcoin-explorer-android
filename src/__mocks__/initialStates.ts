import { BlockType } from "src/components/BlocksInfo/components/Block/type";
import { AddressInfoType, TransactionType } from "src/components/Modal/types";

const timeAgoInitialState = {
  hour: 0,
  minutes: 0,
  day: 0,
  month: 0,
  year: 0,
};

export const initialStateBlocks: BlockType[] = [
  {
    blockHash: "",
    blockHeight: 1,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 2,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 3,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,
    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
  {
    blockHash: "",
    blockHeight: 4,
    satPerVbyte: "1",
    size: "1",
    transactions: 1,
    timeAgo: timeAgoInitialState,

    extras: {
      medianFee: 1,
      pool: {
        name: " ",
      },
    },
  },
];

export const initialStateTransaction: TransactionType = {
  transactionId: " ",
  size: 0,
  fee: 0,
  inputTransactions: [
    {
      prevout: {
        scriptpubkey_address: " ",
        value: 0,
      },
    },
  ],
  outputTransactions: [
    {
      scriptpubkey_address: " ",
      value: 0,
    },
  ],
  statusTransaction: undefined,
};

export const initialStateFormattedTransactions = [
  {
    transactionId: "0",
    value: 0,
    fee: 0,
  },
  {
    transactionId: "1",
    value: 0,
    fee: 0,
  },
];

export const initialStateAddress: AddressInfoType = {
  address: "",
  balance: 0,
  totalAmountReceived: 0,
  totalAmountSent: 0,
};
