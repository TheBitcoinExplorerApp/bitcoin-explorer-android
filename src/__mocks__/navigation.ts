const mockNavigation = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNavigation = jest.requireActual("@react-navigation/native");
  return {
    ...actualNavigation,
    useNavigation: () => ({
      navigate: mockNavigation,
    }),
  };
});
