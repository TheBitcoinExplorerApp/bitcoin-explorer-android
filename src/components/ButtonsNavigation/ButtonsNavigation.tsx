// eslint-disable-next-line import/no-extraneous-dependencies
import { View, Text, TouchableOpacity, StyleSheet } from "react-native";
import useAppDataStore from "src/context/DataProvider";
import HouseIcon from "../../assets/house.svg";
import BlocksIcon from "../../assets/blocks.svg";
import TransactionsIcon from "../../assets/transactions.svg";

type ButtonsNavigationProps = {
  // # TODO: SEARCH FOR A BETTER WAY TO TYPE THIS
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  navigation: any;
  actualScreen: "Home" | "Blocks" | "Transactions";
};

const styles = StyleSheet.create({
  container: {
    backgroundColor: "#111428",
    alignItems: "center",
    flexDirection: "row",
    height: 60,
    justifyContent: "space-evenly",
  },
  buttonContainer: {
    height: "100%",
    alignItems: "center",
    justifyContent: "space-between",
    paddingVertical: 5,
    width: 90,
  },
});

export default function ButtonsNavigation(props: ButtonsNavigationProps) {
  const { navigation, actualScreen } = props;

  const { i18n } = useAppDataStore();

  const homeColor = actualScreen === "Home" ? "#DF7800" : "#D6D6D6";
  const blocksColor = actualScreen === "Blocks" ? "#DF7800" : "#D6D6D6";
  const transactionsColor =
    actualScreen === "Transactions" ? "#DF7800" : "#D6D6D6";

  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => {
          navigation.navigate("Home");
        }}
      >
        <View style={styles.buttonContainer}>
          <HouseIcon style={{ color: homeColor }} />
          <Text
            style={{
              color: homeColor,
            }}
          >
            Home
          </Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => {
          navigation.navigate("Blocks");
        }}
      >
        <View style={styles.buttonContainer}>
          <BlocksIcon style={{ color: blocksColor }} />
          <Text
            style={{
              color: blocksColor,
            }}
          >
            {i18n.t("blocks")}
          </Text>
        </View>
      </TouchableOpacity>

      <TouchableOpacity
        onPress={() => {
          navigation.navigate("Transactions");
        }}
      >
        <View style={styles.buttonContainer}>
          <TransactionsIcon style={{ color: transactionsColor }} />
          <Text
            style={{
              color: transactionsColor,
            }}
          >
            {i18n.t("transactions")}
          </Text>
        </View>
      </TouchableOpacity>
    </View>
  );
}
