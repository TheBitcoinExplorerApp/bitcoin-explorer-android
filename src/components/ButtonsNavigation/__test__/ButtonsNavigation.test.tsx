/* eslint-disable import/no-extraneous-dependencies */
import renderer from "react-test-renderer";
import ButtonsNavigation from "../ButtonsNavigation";
import "src/__mocks__/navigation";

describe("ButtonsNavigation", () => {
  it("renders correctly when at Home", () => {
    const tree = renderer
      .create(<ButtonsNavigation actualScreen="Home" navigation={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when at Blocks", () => {
    const tree = renderer
      .create(<ButtonsNavigation actualScreen="Blocks" navigation={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly when at Transactions", () => {
    const tree = renderer
      .create(
        <ButtonsNavigation actualScreen="Transactions" navigation={() => {}} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
