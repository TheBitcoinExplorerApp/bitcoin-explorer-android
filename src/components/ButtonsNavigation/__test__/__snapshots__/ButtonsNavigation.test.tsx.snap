// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonsNavigation renders correctly when at Blocks 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#111428",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Home
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#DF7800",
          }
        }
      />
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
        Blocos
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Transações
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonsNavigation renders correctly when at Home 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#111428",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#DF7800",
          }
        }
      />
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
        Home
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Blocos
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Transações
      </Text>
    </View>
  </View>
</View>
`;

exports[`ButtonsNavigation renders correctly when at Transactions 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#111428",
      "flexDirection": "row",
      "height": 60,
      "justifyContent": "space-evenly",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Home
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#D6D6D6",
          }
        }
      />
      <Text
        style={
          {
            "color": "#D6D6D6",
          }
        }
      >
        Blocos
      </Text>
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        {
          "alignItems": "center",
          "height": "100%",
          "justifyContent": "space-between",
          "paddingVertical": 5,
          "width": 90,
        }
      }
    >
      <svg
        style={
          {
            "color": "#DF7800",
          }
        }
      />
      <Text
        style={
          {
            "color": "#DF7800",
          }
        }
      >
        Transações
      </Text>
    </View>
  </View>
</View>
`;
